qiskit/tools/apps/optimization.py

```diff
@@ -17,78 +17,94 @@
 """
 Quantum Optimization tools.
 
+
 These are simple tools that are used in our optimization examples
 """
 import sys
 import os
 sys.path.append(os.path.join(os.path.dirname(__file__), '../..'))
 from qiskit import QuantumRegister, ClassicalRegister, QuantumCircuit
-from qiskit.extensions.standard import h, ry, barrier, cz
+from qiskit.extensions.standard import h, ry, barrier, cz, x, y, z
 import numpy as np
-from qiskit.tools.qi.pauli import Pauli, label_to_pauli
-
-
-def SPSA_optimization(obj_fun, initial_theta, SPSA_parameters, max_trials, save_steps = 1,last_avg=1):
-    """Minimize the obj_fun(controls).
-
-    initial_theta = the intial controls
-    SPSA_parameters = the numerical parameters
-    max_trials = the maximum number of trials
+import copy
```

Comment:
Can you move this import behind: import os ?  The recommendation is to group system imports together.